sdk/inc/azure/core/az_json.h

```diff
@@ -60,18 +60,54 @@ typedef struct
 /**
  * @brief Represents a JSON token. The kind field indicates the type of the JSON token and the slice
  * represents the portion of the JSON payload that points to the token value.
+ *
+ * @remarks An instance of #az_json_token must not outlive the lifetime of the #az_json_reader it
+ * came from.
  */
 typedef struct
 {
-  az_json_token_kind kind;
-  az_span slice;
+  az_json_token_kind kind; ///< This read-only field gives access to the type of the token returned
+                           ///< by the #az_json_reader, and it shouldn't be modified by the caller.
+  az_span slice; ///< This read-only field gives access to the slice of the JSON text that
+                 ///< represents the token value, and it shouldn't be modified by the caller.
+                 ///< If the token straddles non-contiguous buffers, this is set to #AZ_SPAN_NULL.
+                 ///< The user can call #az_json_token_copy_into_span() to get the token value into
+                 ///< a contiguous buffer.
+                 ///< In the case of JSON strings, the slice does not include the surrounding
+                 ///< quotes.
+  int32_t size; ///< This read-only field gives access to the size of the JSON text slice that
+                ///< represents the token value, and it shouldn't be modified by the caller. This is
+                ///< useful of the token straddles non-contiguous buffers, to figure out what sized
+                ///< destination buffer to provide when calling #az_json_token_copy_into_span().
 
   struct
   {
     bool string_has_escaped_chars;
+    az_span* pointer_to_first_buffer;
+    int32_t start_buffer_index;
+    int32_t start_buffer_offset;
+    int32_t end_buffer_index;
+    int32_t end_buffer_offset;
   } _internal;
 } az_json_token;
 
+// TODO: Should the parameters be reversed?
```

Comment:
Thoughts on this @JeffreyRichter . The `az_span_copy` API has `destination` first.

Although, I prefer the `source` being first in both cases. Having an `az_json_token` API where `az_json_token` isn't the first parameter is odd.

Reply:
I think what you have is fine as is.